### PR TITLE
chore: introduce a commong klog handling for cmd/nfd-*

### LINKED ISSUE #1339: Common library functions klog config file option handling
**What would you like to be added**:

From https://github.com/kubernetes-sigs/node-feature-discovery/pull/1338#pullrequestreview-1614575394: nfd-master and nfd-worker have virtually identical functionality for handling the klog options (cmdline and config file). It would make sense to try to refactor these into one common library.

**Why is this needed**:

Remove code duplication, less LOC, easier maintenance.

### PR DESCRIPTION
**Pull request by AhmedGrati**
Resolves #1339.

### CODE REVIEW

**Comment by ArangoGutierrez** [comment]
We are in the middle of a release cut, will review once V0.14 is out
Thanks @AhmedGrati for your awesome contribution! 
/hold

**Comment by AhmedGrati** [comment]
@ArangoGutierrez Much appreciated! Also, I've seen that we could externalize handling klog its configuration changes in the config file to `pkg/utils/klog`. wdyt?

File: pkg/utils/klog/klog.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+import (
+	"flag"
+	"strings"
+
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/node-feature-discovery/pkg/utils"
+)
+
+func InitKlogFlags(flagset *flag.FlagSet) map[string]*utils.KlogFlagVal {
```
[the remark this answers is not in the document]
Comment (AhmedGrati) [reply pkg/utils/klog/klog.go:29]: Updated!

File: pkg/utils/klog/klog.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+		klogFlags[name] = utils.NewKlogFlagVal(f)
+		flagset.Var(klogFlags[name], f.Name, f.Usage)
+	})
+
+	return klogFlags
+}
+
+// HandleKlogConfiguration updates the klog configuration taking into consideration
+// klog cli options, and klog configuration specified in the config file.
+func HandleKlogConfiguration(klogArgs map[string]*utils.KlogFlagVal, klogConfig map[string]string) error {
```
Comment (marquiz) [inline pkg/utils/klog/klog.go:45]: Just thinking aloud here: this function merges klog cmdline flags to klog configuration (file) options, basically overriding the config options with anything that was explicitly specified on cmdline. So, should we name this function `UpdateKlogConfigOpts` or `MergeKlogConfiguration` or smth 🤨

Another question: should we specify a separate type for the klog config opts, like
```golang
type KlogConfigOpts map[string]string
```

WDYT?

**Review by marquiz — COMMENTED** [review]
Thanks @AhmedGrati for the PR. Looks pretty clean to me. One comment/question below to think about

File: pkg/utils/klog/klog.go (hunk 2 above)
Comment (AhmedGrati) [reply pkg/utils/klog/klog.go:45]: @marquiz Sounds good to me! Updated!

File: pkg/nfd-master/nfd-master.go [hunk 3]
```diff
@@ -78,7 +80,7 @@ type NFDConfig struct {
 	ResyncPeriod      utils.DurationVal
 	LeaderElection    LeaderElectionConfig
 	NfdApiParallelism int
-	Klog              map[string]string
+	KlogOptions       klogutils.KlogConfigOpts
```
Comment (marquiz) [inline pkg/nfd-master/nfd-master.go:83]: I think we cannot change the name of the field as that is the one used in the actual config file (json/yaml).

```suggestion
	Klog              klogutils.KlogConfigOpts
```

**Review by marquiz — APPROVED** [review]
Good job @AhmedGrati 
/assign @ArangoGutierrez

**Comment by AhmedGrati** [comment]
Thanks for the review @marquiz @ArangoGutierrez.

**Review by ArangoGutierrez — APPROVED** [review]
/lgtm

**Pull request merged**